src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,54 @@
+package duke;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.MemoryFailedException;
+import duke.utils.Parser;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+import duke.utils.Ui;
+
+/**
+ * Represents the entry point for the Duke application. The main function resides in this class.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList allTasks;
+
+    public Duke(String[] memoryPathArray) {
+        this.storage = new Storage(memoryPathArray);
+        this.allTasks = new TaskList();
+        try {
+            this.storage.loadTasks(this.allTasks);
+        } catch (MemoryFailedException e) {
+            Ui.println(e.toString());
+        }
+    }
+
+    /**
+     * Runs the Duke application.
+     */
+    public void run() {
+        Ui.printOnStartup();
+        boolean promptAgain = true;
```

Comment:
Could this boolean be named to shouldPromptAgain?

Reply:
That's true, thank you!